shared/src/commonMain/kotlin/org/hyperskill/app/projects/domain/repository/ProjectsRepository.kt

```diff
@@ -0,0 +1,12 @@
+package org.hyperskill.app.projects.domain.repository
+
+import org.hyperskill.app.projects.domain.model.Project
+
+interface ProjectsRepository {
+    suspend fun getProjects(projectsIds: List<Long>): Result<List<Project>>
+
+    suspend fun getProject(projectId: Long): Result<Project> =
+        kotlin.runCatching {
+            getProjects(listOf(projectId)).getOrThrow().first()
+        }
```

Comment:
It's better to implement this `getProject(projectId: Long)` as extension, because now it is an interface method with default implementation, and it is possible to override it in implementation.

Reply:
@XanderZhu after making this an extension function like this:
```kotlin
interface ProjectsRepository {
    suspend fun getProjects(projectsIds: List<Long>): Result<List<Project>>
}

suspend fun ProjectsRepository.getProject(projectId: Long): Result<Project> =
    kotlin.runCatching {
        getProjects(listOf(projectId)).getOrThrow().first()
    }
```

I got compile time error `Unresolved reference: getProject`:

<img width="498" alt="Screenshot 2023-03-01 at 9 04 26 PM" src="https://user-images.githubusercontent.com/8586063/222147217-5592f099-c45f-4504-aa37-107d4f99f986.png">

Reply:
You simply need to import your extension

Reply:
Ah, my bad, thanks ))